### PR TITLE
Make DisruptedPods in PodDisruptionBudgetStatus optional field

### LINKED ISSUE #63756: When edit PodDisruptionBudget, improper message was produced and cannot updated labels due to required disruptedPods field
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

- PodDisruptionBudget provides inappropriate message and cannot be updated as `disruptedPods` is required filed.

**What you expected to happen**:

- `kubectl edit pdb <PDB_NAME>` works fine for editing labels under `metatdata`. Also, produce correct error message when edit `spec` field.

**How to reproduce it (as minimally and precisely as possible)**:

#### 1. create pdb

~~~
# kubectl create pdb my-pdb --selector=app=nginx --min-available=50%
~~~

#### 2. Edit the pdb

~~~
# kubectl edit pdb my-pdb
// e.g Update label under metadata
  labels:
    foo: bar
  name: my-pdb
~~~

Then, we will get following error after step-2.

~~~
# poddisruptionbudgets.policy "my-pdb" was not valid:
# * : Invalid value: "The edited file failed validation": ValidationError(PodDisruptionBudget.status): missing required field "disruptedPods" in io.k8s.api.policy.v1beta1.PodDisruptionBudgetStatus
~~~

#### Expected:

- When edit `spec`, it should provide correct message as `updates to poddisruptionbudget spec are forbidden.`.

~~~
# poddisruptionbudgets.policy "my-pdb" was not valid:
# * spec: Forbidden: updates to poddisruptionbudget spec are forbidden.
~~~

- When edit `metadata` especially labels under metadata, they should be updated without error.

**Anything else we need to know?**:

According to the code comment, `disruptedPods: null` is valid.

https://github.com/kubernetes/kubernetes/blob/feeee5036321db39fbbf1132c2fc6e31491058a9/pkg/apis/policy/types.go#L71-L73

**Environment**:
- Kubernetes version (use `kubectl version`): build on feeee5036321db39fbbf1132c2fc6e31491058a9

### PR DESCRIPTION
**Pull request by nak3**
**What this PR does / why we need it**:

Please refer to https://github.com/kubernetes/kubernetes/issues/63756

**Which issue(s) this PR fixes**

Fixes https://github.com/kubernetes/kubernetes/issues/63756

**Release note**:

```release-note
`DisruptedPods` field in `PodDisruptionBudget` is optional instead of required.
```

### CODE REVIEW

**Comment by nak3** [comment]
Just rebased.

File: staging/src/k8s.io/api/policy/v1beta1/types.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type PodDisruptionBudgetStatus struct {
 	// the list automatically by PodDisruptionBudget controller after some time.
 	// If everything goes smooth this map should be empty for the most of the time.
 	// Large number of entries in the map may indicate problems with pod deletions.
+	// +optional
 	DisruptedPods map[string]metav1.Time `json:"disruptedPods" protobuf:"bytes,2,rep,name=disruptedPods"`
```
Comment (liggitt) [inline staging/src/k8s.io/api/policy/v1beta1/types.go]: if this is optional, it should also set omitempty. that would mean this field would no longer be serialized as `disruptedPods: null` and would instead be omitted entirely.

File: pkg/apis/policy/types.go [hunk 2]
```diff
@@ -63,6 +63,7 @@ type PodDisruptionBudgetStatus struct {
 	// the list automatically by PodDisruptionBudget controller after some time.
 	// If everything goes smooth this map should be empty for the most of the time.
 	// Large number of entries in the map may indicate problems with pod deletions.
+	// +optional
```
Comment (nikhita) [inline pkg/apis/policy/types.go]: @liggitt is this comment useful for internal types as well?

Comment (liggitt) [reply pkg/apis/policy/types.go]: I can't think of anything that gets generated off the internal types, so I don't think so

**Comment by nak3** [comment]
@liggitt Thank you. Updated.

**Comment by liggitt** [comment]
thanks, please squash down the commits, then LGTM

**Comment by nikhita** [comment]
This should have a release note. @nak3 can you update the main PR body to include a release note?

**Comment by nak3** [comment]
@nikhita Sure, done.

**Pull request merged**